UPGRADING.md

```diff
@@ -116,6 +117,37 @@ method changes in this release to accommodate the different way that
 the new stack tracks data about peers. This change affects users of
 both stacks.
 
+### Using the updated p2p library
+
+The P2P library was reimplemented in this release. The new implementation is
+enabled by default in this version of Tendermint. The legacy implementation is still
+included in this version of Tendermint and can be enabled in the configuration file.
```

Comment:
Maybe a short note that they are cross-compatible

Reply:
Given that the new implementation _is_ compatible, it may be worth mentioning why the old implementation is still around, and when we expect it to go away, e.g.

> The legacy implementation is still included in this version of Tendermint as a backstop to work around unforeseen production issues. If necessary, you can enable the legacy implementation in the server configuration file […] If you need to do this, please consider filing an issue in the Tendermint repository to let us know why.

or words to that effect.